cloudmarker/checks/firewallrulecheck.py

```diff
@@ -0,0 +1,119 @@
+"""Firewall Rule Check plugin for GCP Firewall rule record.
+
+This modules define a :class:`FirewallRuleCheck` that identifies weak Firewall
+rules for Google Cloud Platform.
+"""
+
+
+class FirewallRuleCheck:
+    """Firewall Rule Check plugin.
+
+    This class identifies weak Firewall rules in the record objects by
+    comparing the them with predefined weak Firewall rules
+    configurations.
+    """
+
+    def __init__(self):
+        """Create instance of :class:`FirewallRuleCheck` plugin."""
+        # self._misconfigurations defines the set of weak/misconfigured
+        # firewall rules. Given any firewall rule, is compared with
+        # these predefined weak misconfigurations if the rule matches to
+        # any one predefined rule, it will be marked as weak and will be
+        # reported as anomaly. Adding IPProtocol field have been skipped
+        # to check for misconfiguration as GCP docs says that: The
+        # protocol type is required when creating a firewall rule. This
+        # value can either be one of the following well known protocol
+        # strings (tcp, udp, icmp, esp, ah, ipip, sctp), or the IP
+        # protocol number, implying it cannot be *.
+        self._misconfigurations = {
```

Comment:
Please rename this variable to `self._rules` because this variable is defining the rules to detect misconfigurations.

Reply:
How about `self._weak_rules`? as `_misconfigurations` defines weak rules.

Reply:
The rules in this dictionary are themselves are not weak. 😄 In fact, they are supposed to be *strong* rules that match or detect weak security configurations or security misconfigurations. Keeping this in mind, the names that seem appropriate are: `_detection_rules`, `_misconfiguration_detection_rules`, `_match_rules`, etc. You could choose a name along these lines or simply `_rules`.